pkg/marathon/client.go

```diff
@@ -31,18 +49,87 @@ func NewClient(ctx api.Context) (*Client, error) {
 		baseURL = cluster.URL() + "/service/marathon"
 	}
 
-	config := marathon.NewDefaultConfig()
+	config := goMarathon.NewDefaultConfig()
 	config.URL = baseURL
 	config.HTTPClient = pluginutil.NewHTTPClient()
 
-	client, err := marathon.NewClient(config)
+	client, err := goMarathon.NewClient(config)
 	if err != nil {
 		return nil, err
 	}
 
 	return &Client{API: client, baseURL: baseURL}, nil
 }
 
+// AddApp creates a deployment from the app definition referenced in appFile which can be a local
+// file name or an HTTP URL. If appFile is empty, the definition is read from ctx.Input().
+func (c *Client) AddApp(ctx api.Context, appLocation string) (*goMarathon.Application, error) {
+	appBytes, err := getAppDefinition(ctx, appLocation)
+	if err != nil {
+		return nil, err
+	}
+	var app goMarathon.Application
+	err = json.Unmarshal(appBytes, &app)
+	if err != nil {
+		return nil, err
+	}
+
+	existingApp, err := c.API.ApplicationBy(app.ID, &goMarathon.GetAppOpts{})
+	if err != nil {
+		if apiErr, ok := err.(*goMarathon.APIError); !ok {
+			return nil, err
+		} else if apiErr.ErrCode != goMarathon.ErrCodeNotFound {
+			return nil, err
+		}
+	}
+	if existingApp != nil {
+		return nil, ErrAppAlreadyExists{appID: app.ID}
+	}
+
+	return c.API.CreateApplication(&app)
+}
+
+// getAppDefinition loads the app definition JSON from either a file pointed to by location or an HTTP
+// URL pointed to by location or, if location is empty, from stdin.
+func getAppDefinition(ctx api.Context, location string) ([]byte, error) {
+	var appBytes []byte
+	var err error
+
+	if location == "" {
+		appBytes, err = ioutil.ReadAll(ctx.Input())
+		if err != nil {
+			return nil, err
+		}
+		return appBytes, nil
+	}
+
+	locationURL, err := url.Parse(location)
+	if err != nil {
+		return nil, err
+	}
+
+	switch locationURL.Scheme {
+	case "":
+		appBytes, err = ioutil.ReadFile(location)
+		if err == nil {
+			return appBytes, nil
+		}
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+	case "http", "https":
```

Comment:
Does the python version require scheme?

Reply:
Implicitly, yes. It only supports HTTP: https://github.com/dcos/dcos-core-cli/blob/86fe4f9c213d597adbc8daf9c0a672c871b24711/python/lib/dcoscli/dcoscli/marathon/main.py#L306